src/linuxcan.cpp

```diff
@@ -28,49 +29,46 @@ KvaserCan::~KvaserCan()
     canClose(*handle);
 }
 
-ReturnStatuses KvaserCan::open(const int32_t& hardware_id,
-                               const int32_t& circuit_id,
-                               const int32_t& bitrate,
+ReturnStatuses KvaserCan::open(const uint64_t& hardware_id,
```

Comment:
Does the hardware_id need to be 64 bit? It's a six digit decimal number on all the kvaser devices I've seen and log2(10^7) is just over 23

Reply:
The canlib API specifies that it's an 8-byte (64-bit) value. See https://www.kvaser.com/canlib-webhelp/canlib_8h.html#aba2148fef4de2c724510cead2edd4060